utils/documentOps.py

```diff
@@ -157,18 +154,23 @@ def parse_file(file_name):
         if p is not None:
             assert op is not None, "Min without op."
             assert op not in min_dict, "Redefinition of min for op " + op
-            min_dict[op] = int(p[1])
+            op_min = max(int(p[1]), min_opset_default)
+            min_dict[op] = op_min
+            # By default, assume we support the latest op version. (overridden by UNSUPPORTED)
+            max_dict[op] = max_opset_default
             if debug:
                 print("Got min for op", op, ":", min_dict[op])
             continue
         # Max release supported.
-        p = re.search(r'==MAX==\s+(\d+)\s*$', l)
+        p = re.search(r'==UNSUPPORTED==\s+(.*)\s*$', l)
         if p is not None:
-            assert op is not None, "Max without op."
-            assert op not in max_dict, "Redefinition of max for op " + op
-            max_dict[op] = int(p[1])
+            assert op is not None, "Unsupported without op."
+            assert op in min_dict and min_dict[op], "Unsupported without min for op " + op
+            assert max_dict[op] == max_opset_default, "Redefinition of Unsupported for op " + op
```

Comment:
This support is getting a bit funky, as you have a different behavior if no min and unsupported is given.

Maybe cleaner would be to set min/max by default when op is defined (processing of `==OP==`.
That way, as soon as an op is defined, its min and max are defined, and then we may redefine them by having `MIN` or `UNSUPPORTED`
 
For completion, you may want to check when setting either min or max that `min<=max`

Reply:
> This support is getting a bit funky, as you have a different behavior if no min and unsupported is given.

That was by design. Based on how it's used, `==UNSUPPORTED==` only makes sense if there's a `==MIN==`. We could see about renaming UNSUPPORTED to something that's more obvious that it shouldn't exist without a MIN (UNSUPPORTED_AT_VER?). However the end result is still the same; that key doesn't make sense without a MIN.

It's not dissimiar to how `==MIN==` is handled. It is an error if there's a `==MIN==` without an `==OP==` before it.

> Maybe cleaner would be to set min/max by default when op is defined.

I already do this for the max with `max_dict[op] = max_opset_default`. By default it'll get a `*` as the max by default. If an ==UNSUPPORTED== is specified, then the default max is overridden.

Prior to my changes, if there was an `==OP==` it would show as unsupported unless a `==MIN==` was specified. I assumed this was intentional to force a dev to specify lowest `==MIN==` their addition supported. So the error I added was to help catch a semantic error (ie `==OP==` with `==UNSUPPORTED==` but no `==MIN==`) which otherwise would just doc the Op as unsupported.

> For completion, you may want to check when setting either min or max that min<=max

Good idea. I'll include this in the next update.